### PR TITLE
Add Series.shift() (with minor cleanups)

### PR DESCRIPTION
**Pull request by HyukjinKwon**
This PR add shift() in Series too.

### CODE REVIEW

File: docs/source/reference/frame.rst [hunk 1]
```diff
@@ -174,6 +174,16 @@ Combining / joining / merging
...
 
+Time series-related
+-------------------
+.. autosummary::
+   :toctree: api/
+
+   DataFrame.shift
+
+.. autosummary::
+   :toctree: api/
```
Comment (ueshin) [inline docs/source/reference/frame.rst]: Do we need this?

File: databricks/koalas/base.py [hunk 2]
```diff
@@ -520,3 +520,66 @@ def any(self, axis: Union[int, str] = 0) -> bool:
...
+
+        Returns
+        -------
+        Copy of input Series/Index, shifted.
+
+        Examples
+        --------
+        >>> df = ks.DataFrame({'Col1': [10, 20, 15, 30, 45],
+        ...                    'Col2': [13, 23, 18, 33, 48],
+        ...                    'Col3': [17, 27, 22, 37, 52]})
```
Comment (ueshin) [inline databricks/koalas/base.py]: This equivalent example in `DataFrame` needs `columns=[...]`. Could you fix it too?

**Comment by HyukjinKwon** [comment]
Thanks, @ueshin!

File: databricks/koalas/base.py (hunk 2 above)
Comment (ueshin) [reply databricks/koalas/base.py]: Could you also add it to https://github.com/databricks/koalas/blob/a248cc6788e294aebf03d9344429872512cd1db4/databricks/koalas/frame.py#L1901-L1904 as well?

Comment (HyukjinKwon) [reply databricks/koalas/base.py]: Yup!

**Pull request merged**